### PR TITLE
refactor text cleaning steps

### LINKED ISSUE #41: Move keyword analysis data cleaning steps into the raw_to_text notebook
As a Data Scientist I expect the text data set to already be cleaned, so that I don't have to worry about additional data cleaning in my analysis notebooks.

Currently the data cleaning step for the bulk of the email body text occurs in "notebooks/02_analyses/keywords" it should be moved to the raw_to_text notebook. 

Acceptance Criteria:

- [ ]  Data Cleaning step is removed from keywords notebook
- [ ]  Ceph bucket `text` dir is updated with clean data set

### PR DESCRIPTION
**Pull request by MichaelClifford**
## Related Issues and Dependencies

#41 

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

This PR refactors the data cleaning steps so that it occurs in the `raw_to_text` notebooks instead of `keywords analysis` notebook. This ensures that the data we are storing on ceph in already cleaned, and we are using the same text cleaning methods for all our analysis notebooks.

### CODE REVIEW

**Review by oindrillac — APPROVED** [review]
🚢

File: notebooks/01_collect_data/raw_to_text.ipynb [hunk 1]
```diff
@@ -17,28 +17,27 @@
   },
```
Comment (tumido) [inline notebooks/01_collect_data/raw_to_text.ipynb:17]: <p>I know you're just moving code around, but why not to take a look at it. :D</p><p> </p><p>Maybe not directly useful in this particular case:</p><p> </p><p>Python has a built in <code>calendar</code> module, so you can get <a href="https://docs.python.org/3/library/calendar.html#calendar.day_name" target="_blank">month and weekday names</a> without hard-coding stuff</p><p> </p><pre class="ql-syntax" spellcheck="false">&gt;&gt;&gt; import calendar

&gt;&gt;&gt; [i for i in calendar.month_name]
['', 'January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December']

&gt;&gt;&gt; [i for i in calendar.month_abbr]
['', 'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']

&gt;&gt;&gt; [i for i in calendar.day_name]
['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']

&gt;&gt;&gt; [i for i in calendar.day_abbr]
['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']
</pre><p>I'd also suggest to make the regex checks case insensitive.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/mailing-list-analysis-toolkit/pull/64/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

**Comment by MichaelClifford** [comment]
@tumido good call at taking a bit closer look at these regex : ) 

I think there is some additional improvement we can be doing with these cleaning steps overall (looks like some of these are actually removing more data then they intend to). I think it might makes sense to make it a separate issue, to determine exactly what cases are and aren't being caught by the existing regex and write some tests to confirm instead of "eye-balling" the output or writing in-notebook cases on the fly.

And as far as using `calendar` here, also a good note for reference, but do you think programtically constructing regex statements is the best way to go? I would think using a more complete regex statement like `((mon|tues|wed(nes)?|thur(s)?|fri|sat(ur)?|sun)(day)?)` and something similar for the months might be better to reduce the number of `re.match` statements used and possible be more readable then list comprehensions nested in regex. WDYT?

**Comment by Shreyanand** [comment]
For readability and optimization, you could also collect all the regex in one place (this notebook or a separate text file) and compile them (re.compile) and then apply it on the lines...

**Comment by tumido** [comment]
I can't agree more with both of you. :slightly_smiling_face: 

When it comes to the `calendar` module usage - for me that was more of a note that this module exists. I think it's often forgotten and overlooked module with some neat functions if you need to work with dates. :slightly_smiling_face: So.. I wanted to promote it a bit. :smile: It may not be the best fit for this case, you're right.

Filing the issue for ya. :+1:

**Comment by MichaelClifford** [comment]
:+1:  @tumido Should we merge this PR as is then? and move additional changes to the linked issue?

**Review by tumido — APPROVED** [review]
@MichaelClifford I think that's right way to approach this.

/lgtm (abbrev for "Let's get this merged", lol)
/approve

**Review by Shreyanand — APPROVED** [review]
lgtm :rocket:

**Pull request merged**